### PR TITLE
feature/volume handler parser

### PR DESCRIPTION
**Pull request by harishp8889**
# Description
Implemented Volume handler parser for the volume group snapshot

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/268 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Tested in Kubernetes cluster by creating a volume group snapshot

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,20 +7,21 @@ module github.com/dell/csi-vxflexos/v2
 
 //replace github.com/dell/goscaleio => ./goscaleio
 
-replace github.com/dell/gocsi => ./gocsi
+// replace github.com/dell/gocsi => ./gocsi
```
Comment (JacobGros) [inline go.mod:10]: Can we put this line back in?
We modified gocsi; so building/testing should be done with the latest

File: service/csi_extension_server.go [hunk 2]
```diff
@@ -434,3 +435,14 @@ func (s *service) buildCreateVGSResponse(ctx context.Context, snapResponse *siot
 	return groupSnapshots, nil
 
 }
+
+func (s *service) ParseVolumeHandle(ctx context.Context, request *volumeGroupSnapshot.VolumeHandleRequest) (*volumeGroupSnapshot.VolumeHandleResponse, error) {
+	if request.VolumeHandle == "" {
+		return nil, errors.New("volume handle should not be empty")
+	}
+	volHandle := strings.Split(request.VolumeHandle, "-")
```
Comment (JacobGros) [inline service/csi_extension_server.go:443]: We used similar logic and ran into issues where ArrayID was actually a system name with hyphens.
@nb950, is that a possible issue here as well?

Could implement something similar to [this](https://github.com/dell/csi-powerflex/blob/be5bedda3fd00f29cc1238fbe69b22f6c180991e/service/service.go#L664)

Comment (harishp8889) [reply service/csi_extension_server.go:443]: @JacobGros This is the similar thing done as of now in the sidecar, I just took the same thing and added it in this parser function https://github.com/dell/csi-volumegroup-snapshotter/blob/main/controllers/dellcsivolumegroupsnapshot_controller.go#L861

File: go.mod (hunk 1 above)
Comment (harishp8889) [reply go.mod]: done

File: service/csi_extension_server.go (hunk 2 above)
Comment (harishp8889) [reply service/csi_extension_server.go:443]: Probably inside this parser RPC we can just call  `getVolumeIDFromCsiVolumeID` and ` getSystemIDFromCsiVolumeID`  methods by passing the entire volume handle. @nb950 @JacobGros Please let me know if this works. Thank you

Comment (JacobGros) [reply service/csi_extension_server.go:443]: sounds good to me

Comment (harishp8889) [reply service/csi_extension_server.go]: Done

**Review by JacobGros — APPROVED** [review]
LGTM

**Review by JacobGros — CHANGES_REQUESTED** [review]
Let's meet w/Jai to go over this first

**Comment by harishp8889** [comment]
> Let's meet w/Jai to go over this first

Had a call with @nb950, This PR is no longer required as the parsing logic is added locally at the sidecar itself and hence closing this PR.

**Pull request closed without merging**